src/Countries/Malaysia.php

```diff
@@ -0,0 +1,562 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+use DateTime;
+use IntlDateFormatter;
+use Spatie\Holidays\Exceptions\InvalidRegion;
+use Spatie\Holidays\Exceptions\InvalidYear;
+
+class Malaysia extends Country
+{
+    protected string $timezone = 'Asia/Kuala_Lumpur';
+
+    /** @var array<int, string> $regions */
+    protected array $regions = [
+        'jhr',
+        'kdh',
+        'ktn',
+        'kul',
+        'lbn',
+        'mlk',
+        'nsn',
+        'phg',
+        'png',
+        'prk',
+        'pls',
+        'pjy',
+        'sbh',
+        'swk',
+        'sgr',
+        'trg',
+    ];
+
+    protected function __construct(
+        protected ?string $region = null,
+    ) {
+    }
+
+    public function countryCode(): string
+    {
+        return 'my';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge(
+            $this->fixHolidays($year),
+            $this->variableHolidays($year)
+        );
+    }
+
+    /**
+     * @param int $year
+     * @return array<string, string>
+     */
+    protected function fixHolidays(int $year): array
+    {
+        return [
+            'Hari Pekerja' => '05-01',
+            'Hari Kebangsaan' => '08-31',
+            'Hari Malaysia' => '09-16',
+            'Hari Krismas' => '12-25',
+        ];
+    }
+
+    /**
+     * @param int $year
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function variableHolidays(int $year): array
+    {
+        $variableHolidays = [
```

Comment:
It's hard to get a grip of this. Could you format this in a more readable way? You can group the chinese related holidays together in a dedicated method, the same for the islamic related holidays.